src/main/java/me/crespel/runtastic/mapper/GpxSportSessionMapper.java

```diff
@@ -64,12 +86,56 @@ public GpxType mapSportSession(SportSession session, String format) {
 			}
 			trk.getTrkseg().add(trkseg);
 		}
+		gpx.getTrk().add(trk);
 
-		GpxType gpx = factory.createGpxType();
-		gpx.setVersion("1.1");
-		gpx.setCreator("RuntasticExportConverter");
+		if (session.getGpx() != null ) {
+			// handling GPX GPS data; add them to first /trk as /trkseg
+			gpx.getTrk().get(0).getTrkseg().addAll(session.getGpx().getTrk().get(0).getTrkseg());

```

Comment:
Not entirely sure about this, but won't this add a _second_ trkseg, identical to the one already added from the JSON GPS data? if so, you might simply want to use GPX data directly if it's available, and if not use JSON data.

Also, you might want to add an extra check in case the GPX does NOT contain a trkseg (activity with no GPS data), if that's possible.

Reply:
Fixed. Now the handling of "JSON GPS data" and "GPX GPS data" is better separated.